concent_api/gatekeeper/views.py

```diff
@@ -56,6 +62,10 @@ def upload(request):
 @require_safe
 def download(request):
     logger.debug("Download request received.")
+
+    if config.SHUTDOWN_MODE is True and not is_request_from_concent_api(request):
+        return JsonResponse({'error': 'Concent is in shutdown mode.'}, status=503)
```

Comment:
Have you checked what nginx-storage will return if you return HTTP 503 here? Keep in mind that this JSON message does not go directly to the client. It goes to nginx and [ngx_http_auth_request](https://nginx.org/en/docs/http/ngx_http_auth_request_module.html) module decides what to return. The docs say that it expects either HTTP 200 or HTTP 401. We need to check what happens if you return HTTP 503. Will it return it or will it replace it with a HTTP 500 saying that it received something unexpected? Ask @bartoszbetka for help in checking that on the cluster.

Reply:
No, I didn't, but it would be good to check.

@bartoszbetka can you help with that ?

Reply:
@bartoszbetka says that nginx returns HTTP 500. So this needs to be changed.

Reply:
This can't easily be done in nginx config. We'd have to stop using the module and rewrite this logic in lua which is not what we really want to do.

Return HTTP 401.

Reply:
done :+1: